### ISSUE TITLE
DeviceInfo.getMostRecentLocation crashes the app

### ISSUE THREAD

**Issue by ipereziriarte** [amplitude/Amplitude-Android#104]
Hi,

I'm seeing inside the production logs a small amount of crashes.
Looks like **FORCE_STOP_PACKAGES permission is missing**

### StackTrace

```
Fatal Exception: java.lang.SecurityException: Permission Denial: forceStopPackage() from pid=21114, uid=10091 requires android.permission.FORCE_STOP_PACKAGES
       at android.os.Parcel.readException(Parcel.java:1425)
       at android.os.Parcel.readException(Parcel.java:1379)
       at android.location.ILocationManager$Stub$Proxy.getProviders(ILocationManager.java:851)
       at android.location.LocationManager.getProviders(LocationManager.java:321)
       at com.amplitude.api.DeviceInfo.getMostRecentLocation(DeviceInfo.java:309)
       at com.amplitude.api.DeviceInfo$CachedInfo.getCountryFromLocation(DeviceInfo.java:127)
       at com.amplitude.api.DeviceInfo$CachedInfo.getCountry(DeviceInfo.java:110)
       at com.amplitude.api.DeviceInfo$CachedInfo.(DeviceInfo.java)
       at com.amplitude.api.DeviceInfo$CachedInfo.(DeviceInfo.java)
       at com.amplitude.api.DeviceInfo.getCachedInfo(DeviceInfo.java:237)
       at com.amplitude.api.DeviceInfo.prefetch(DeviceInfo.java:243)
       at com.amplitude.api.AmplitudeClient$1.run(AmplitudeClient.java:163)
       at android.os.Handler.handleCallback(Handler.java:725)
       at android.os.Handler.dispatchMessage(Handler.java:92)
       at android.os.Looper.loop(Looper.java:153)
       at android.os.HandlerThread.run(HandlerThread.java:60)
```

### Environment

Amplitude version 2.7.1
Affected environment: production
Android version: Android 4.x and Android 5.x
Devices: 
+ huawei g510_0200
+ samsung_sgh_i527
+ samsung gt_i9060i
+ samsung sm_g530t1
+ vivo 4_8 hd

Thanks for your support.

**Comment by djih** [comment]
@ipereziriarte does this crash only occur on those 5 devices? I wonder if it's a device-specific issue, like their manufacturer restricts access to GPS

**Comment by djih** [comment]
@ipereziriarte on our SDK we can catch security exceptions when fetching location to prevent crashes, but I'm curious as to why it's asking for that permission? That permission is the ability for your app to force close other apps. Fetching the location information should not require that permission. Here are some posts I found: 

1. http://stackoverflow.com/questions/20717708/android-permission-denial-forcestoppackage
2. http://stackoverflow.com/questions/6577591/activitymanager-forcestoppackage-from-task-manager/6578213#6578213 

The solution for this is to add that permission to your app's `AndroidManifest.xml`, and have your app be signed. Any ideas? None of our other customers have reported this issue.

**Comment by ipereziriarte** [comment]
@djih is really weird. As I told you in the ticket, is happening to a really small amount of users.
It's happening on a few more devices but their appear under the "other" tab in Fabric so I can't see which are them. 

We are not trying to force close any other app from inside our app. Also, I've checked the merged manifest, and none of the 3rd party libraries that we are using requests that permission.

Maybe it's ok to live with it as it happening to a really low amount of users.

**Comment by djih** [comment]
@ipereziriarte https://github.com/amplitude/Amplitude-Android/pull/107 has been released as v2.7.2, it might take a few hours for it to appear in Maven

**Comment by ipereziriarte** [comment]
Thanks!
